File: leetcodeGo/greedy/BestTimeToBuyAndSellStock2.go
func maxProfit(prices []int) int {
    var maxProfit int
    l, r := 0, 1
    for {
        if r >= len(prices) {
            break
        }
        if prices[r] > prices[l] {
            maxProfit += prices[r] - prices[l]
        }
        r += 1
        l += 1
    }
    return maxProfit
}

//TC, SC = O(n)

File: leetcode_rust/greedy/BestTimeToBuyAndSellStock2.rs
impl Solution {
    pub fn max_profit(prices: Vec<i32>) -> i32 {
        let mut maxProfit = 0;
        for i in 1..prices.len() {
            if prices[i] > prices[i-1] {
                maxProfit += prices[i] - prices[i-1];
            }
        }
        maxProfit
    }
}

//TC, SC: O(n)
